auction-app-frontend/src/services/productService.js

```diff
@@ -14,4 +14,12 @@ const getProductRandom = () => {
   return getRequest("/products/random");
 };
 
-export { getProductsPaginated, getProduct, getProductRandom };
+const getProductsPaginated = (page, size, categoryId, searchQuery) => {
+  // construct query string based on the presence of categoryId or search query
+  const categoryParam = categoryId ? `&category_id=${categoryId}` : '';
+  const searchParam = searchQuery ? `&search_product=${encodeURIComponent(searchQuery)}` : '';
+  
+  return getRequest(`/products?page=${page}&size=${size}${categoryParam}${searchParam}`);
+};
+
+export { getProductsPaginated, getProduct, getProductRandom, getProductsPaginatedSorted };
```

Comment:
Why are we exporting `getProductsPaginated` and `getProductsPaginatedSorted`, I don't see both methods here.
Also `getProducts` it's completely fine, pagination and sorting are expected, no need for it in the name.